### PR TITLE
ADD: If unable to connect to server, show alert.

### PR DESCRIPTION
**Pull request by marcosrdz**
Screen Shot 2021-01-22 at 10 09 56 PM

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/x84p050d6mqykczrwtv0n8hdjm

[download apk](http://filestorage.bluewallet.io:1488/connectalert-1611372267.apk)

**Comment by limpbrains** [comment]
@marcosrdz What if I haven't changed electrum settings and BW still can't connect, will this error appear to me ?

**Comment by marcosrdz** [comment]
> @marcosrdz What if I haven't changed electrum settings and BW still can't connect, will this error appear to me ?

Yes. It will. If after 5 attempts the connections fails the alert is shown.

**Comment by limpbrains** [comment]
What's the point of showing it, if `Reset to default` will not help me?

**Comment by marcosrdz** [comment]
Lets you know "hey...your internet is not in good enough condition or ur server is unreachable"

**Comment by limpbrains** [comment]
I've put my device offline, alert popped up, I've pressed Cancel. Now I've turned internet connection on, but BW still can't connect. It doesn't update on pull-to-refresh

**Comment by marcosrdz** [comment]
> I've put my device offline, alert popped up, I've pressed Cancel. Now I've turned internet connection on, but BW still can't connect. It doesn't update on pull-to-refresh

Hmmm ok. Ill take a look

**Review by Overtorment — APPROVED** [review]
lgtm

**Comment by Overtorment** [comment]
@marcosrdz  @limpbrains does this need more work?

**Comment by limpbrains** [comment]
I believe this issue needs to be checked/fixed 

> I've put my device offline, alert popped up, I've pressed Cancel. Now I've turned internet connection on, but BW still can't connect. It doesn't update on pull-to-refresh

**Comment by marcosrdz** [comment]
> I believe this issue needs to be checked/fixed
> 
> > I've put my device offline, alert popped up, I've pressed Cancel. Now I've turned internet connection on, but BW still can't connect. It doesn't update on pull-to-refresh

Updated. Please review.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/v786rhf187yw82jd6ttkg9tg3g

[download apk](http://filestorage.bluewallet.io:1488/connectalert-1614317749.apk)

File: blue_modules/BlueElectrum.js [hunk 1]
```diff
@@ -96,13 +99,89 @@ async function connectMain() {
...
+    loc.formatString(
+      usingPeer ? loc.settings.electrum_unable_to_connect : loc.settings.electrum_error_connect,
+      usingPeer ? { server: `${usingPeer.host}:${usingPeer.ssl ?? usingPeer.tcp}` } : {},
+    ),
+    [
+      {
+        text: loc.wallets.list_tryagain,
+        onPress: () => {
+          connectionAttempt = 0;
+          mainClient.close() && mainClient.close();
```
Comment (limpbrains) [inline blue_modules/BlueElectrum.js:126]: Why call `.close()` two times?

Comment (marcosrdz) [reply blue_modules/BlueElectrum.js:126]: idk the reason. i simply copied whats currently in use. @Overtorment  should know the reason behind this.

**Review by limpbrains — APPROVED** [review]
>  I've put my device offline, alert popped up, I've pressed Cancel. Now I've turned internet connection on, but BW still can't connect. It doesn't update on pull-to-refresh

My issue has been fixed

File: blue_modules/BlueElectrum.js (hunk 1 above)
Comment (Overtorment) [reply blue_modules/BlueElectrum.js:126]: iForgot

**Pull request merged**